### PR TITLE
Refactor current card API

### PR DESCRIPTION
**Pull request by freddidierRTE**
The card api is now just an interface to CardTemplateGateway that acts
as a bridge between the template code and the angular card components code

Nothing in release note

### CODE REVIEW

File: ui/main/src/app/business/store/currentDisplayedCard.store.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    private static _userMemberOfAnEntityRequiredToRespond: boolean;
+    public static init() {
+        CurrentDisplayedCardStore._card = undefined;
+        CurrentDisplayedCardStore._childCards = [];
+        CurrentDisplayedCardStore._displayContext = '';
+        CurrentDisplayedCardStore._entitiesAllowedToRespond = [];
+        CurrentDisplayedCardStore._entitiesUsableForUserResponse = [];
+        CurrentDisplayedCardStore._responseLocked = false;
+        CurrentDisplayedCardStore._userAllowedToRespond = false;
+    }
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/business/store/currentDisplayedCard.store.ts]: ok

File: ui/main/src/app/business/store/currentDisplayedCard.store.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    public static setEntitiesAllowedToRespond(entitiesAllowedToRespond: string[]) {
+        CurrentDisplayedCardStore._entitiesAllowedToRespond = entitiesAllowedToRespond;
+    }
+    public static getEntitiesUsableForUserResponse() {
+        return CurrentDisplayedCardStore._entitiesUsableForUserResponse;
+    }
+    public static setEntitiesUsableForUserResponse(entitiesUsableForUserResponse: string[]) {
+        CurrentDisplayedCardStore._entitiesUsableForUserResponse = entitiesUsableForUserResponse;
+    }
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/business/store/currentDisplayedCard.store.ts]: ok

File: ui/main/src/app/business/store/currentDisplayedCard.store.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    public static setChildCards(value: Card[]) {
+        CurrentDisplayedCardStore._childCards = value;
+    }
+    public static getCard(): Card {
+        return CurrentDisplayedCardStore._card;
+    }
+    public static setCard(card: Card) {
+        CurrentDisplayedCardStore._card = card;
+    }
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/business/store/currentDisplayedCard.store.ts]: ok

File: ui/main/src/app/business/store/currentDisplayedCard.store.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    public static getEntitiesUsableForUserResponse() {
+        return CurrentDisplayedCardStore._entitiesUsableForUserResponse;
+    }
+    public static setEntitiesUsableForUserResponse(entitiesUsableForUserResponse: string[]) {
+        CurrentDisplayedCardStore._entitiesUsableForUserResponse = entitiesUsableForUserResponse;
+    }
+    public static isResponseLocked() {
+        return CurrentDisplayedCardStore._responseLocked;
+    }
+    public static setReponseLocked(responseLocked: boolean) {
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/business/store/currentDisplayedCard.store.ts]: ok

File: ui/main/src/app/business/builtInTemplates/message-or-question-list/usercard/message-or-question-listUserCardTemplateView.spec.ts [hunk 5]
```diff
@@ -53,15 +55,15 @@ describe('MessageOrQuestionList UserCard template', () => {
     it('GIVEN an existing card WHEN user edit card THEN message/summary are actual message/summary', () => {
         const view = new MessageOrQuestionListUserCardTemplateView();
         CurrentUserCardAPI.currentUserCard.editionMode = 'EDITION';
-        CurrentCardAPI.currentCard.card = {data: {richMessage: 'My message', summary: 'My summary'}};
+        CurrentDisplayedCardStore.setCard({data: {richMessage: 'My message', summary: 'My summary'}} as Card);
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/business/builtInTemplates/message-or-question-list/usercard/message-or-question-listUserCardTemplateView.spec.ts]: done in a future PR

File: ui/main/src/app/modules/share/template-rendering/template-rendering.component.ts [hunk 6]
```diff
@@ -87,31 +87,31 @@ export class TemplateRenderingComponent implements OnChanges, OnInit, OnDestroy,
...
             that.isLoadingSpinnerToDisplay = true;
             that.changeDetector.markForCheck();
-        };
-        CurrentCardAPI.currentCard.hideLoadingSpinner = function () {
+        });
+        CardTemplateGateway.setFunctionToHideLoadingSpinner(() => {
             that.isLoadingSpinnerToDisplay = false;
             that.changeDetector.markForCheck();
-        };
+        });
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/modules/share/template-rendering/template-rendering.component.ts]: ok

File: ui/main/src/app/business/templateGateway/cardTemplateGateway.ts [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+    private static _childCards: Card[];
+    private static _displayContext: string;
+    private static _entitiesAllowedToRespond: string[];
+    private static _entitiesUsableForUserResponse: string[];
+    private static _responseLocked: boolean;
+    private static _userAllowedToRespond: boolean;
+    private static _userMemberOfAnEntityRequiredToRespond: boolean;
+
+    private static _functionToDisplayLoadingSpinnner: Function;
+    private static _functionToHideLoadingSpinnner: Function;
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/business/templateGateway/cardTemplateGateway.ts]: ok

**Review by ClementBouvierN — APPROVED** [review]

**Pull request merged**